h/search/core.py

```diff
@@ -34,14 +35,25 @@ class Search(object):
     :type stats: statsd.client.StatsClient
     """
     def __init__(self, request, separate_replies=False, stats=None, _replies_limit=200):
-        self.request = request
-        self.es = request.es
+        self.es_index = request.es.index
         self.separate_replies = separate_replies
         self.stats = stats
         self._replies_limit = _replies_limit
-
-        self.builder = self._default_querybuilder(request, self.es)
-        self.reply_builder = self._default_querybuilder(request, self.es)
+        # Order matters!!! The KeyValueMatcher must be run last,
+        # after all other qualifiers have popped off the params.
+        self._qualifiers = [query.Sorter(),
+                            query.Limiter(),
+                            query.DeletedFilter(),
+                            query.AuthFilter(request),
+                            query.UriFilter(request),
+                            query.GroupFilter(),
+                            query.GroupAuthFilter(request),
+                            query.UserFilter(),
+                            query.NipsaFilter(request),
+                            query.AnyMatcher(),
+                            query.TagsMatcher(),
+                            query.KeyValueMatcher()]
```

Comment:
This KeyValueMatcher matches any leftovers-before it was hidden in the depth of the Builder class and now it's more obvious what's going on. Although this catchall thing that has to be run always at the end is kinda awkward. Maybe the improvements we talked about to the api will make this go away?